src/cache/cache-localstorage.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  CacheKey,
+  findExistingCacheKey,
+  ICache,
+  CACHE_KEY_PREFIX
+} from './shared';
+
+export class LocalStorageCache implements ICache {
+  public set(key: string, entry: unknown): Promise<void> {
+    localStorage.setItem(key, JSON.stringify(entry));
+    return Promise.resolve();
+  }
+
+  public get(key: string): Promise<unknown> {
```

Comment:
Is there any reason this has to be marked with returning Unknown? Would it make sense to leverage generics here and use

```ts
public get<TData = unknown>(key: string) : Promise<TData> {
  ...
  const payload = this.readJson<TData>(cacheKey);
  ...
}

private readJson<TData>(cacheKey: CacheKey): TData {
}
```